### PR TITLE
Resolve configuration errors

### PR DESCRIPTION
**Pull request by debelistic**
### What does this PR do?

- set sequelizerc for transpiling to native JS
- set sequelize scripts
- set development environment executable
- seed admin user to database using environment secrete
- remove production configuration errors

### Description of Task to be completed?

- add @babel/register to sequelize configuration
- use coveralls badge in README
- add scripts for database migrations and seeders
- setup tasks running in development environment
- remove devdependencies from production executable file
- add admin status to User model
- add admin seeder

### How should this be manually tested?

- run `npm run reset-db` to drop and migrate database
- run `npm run seeders` to add admin, ensure environment secrete is set in `.env` file
- run `npm run devstart` to see app in development environment
- run `npm start` to see app in production environment

### Any background context you want to provide?

The `dev ` file is, so we can run development execute file independently of the `www` file. In the `www` file we are making use of the compiled files from our `dist` folder, while the `dev` file makes use of ES6 code from `src` folder, and we require babel dependencies at the beginning so it can compile our code. 

The reason for this is that;
-  the current `www` requires babel which is a devDependency, and app would crash in production since Heroku only install project dependencies
-  we could move the babel dependencies to production as we are doing presently, however compiling code in production is inefficient.

### What are the relevant pivotal tracker stories?
#168250646

### Implementation Plan

N/A

### Questions:

- suggest any other config errors or conventions to be resolved in your review

### CODE REVIEW

**Comment by voke1** [comment]
Good work @debelistic However, I got this error when I ran `npm run test` ```ERROR: no pg_hba.conf entry for host "129.56.74.94", user "uahtduhrsmfqxc", database "dfrvfn41s4u850", SSL off```. 

I had to set ssl to true by specifying this ```dialectOptions: {
      ssl: true,
    }``` in the `config.js` file under both test and development for it to work. You may want to check that.


I also get this when I ran `npm run seeders` 

``` 
ERROR: Validation error
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! express-authorshaven@1.0.0 seeders: `sequelize db:seed:all`
npm ERR! Exit status 1
``` 
I'm presuming this might have to do with the seeder file

**Comment by debelistic** [comment]
> Good work @debelistic However, I got this error when I ran `npm run test` `ERROR: no pg_hba.conf entry for host "129.56.74.94", user "uahtduhrsmfqxc", database "dfrvfn41s4u850", SSL off`.
> 
> I had to set ssl to true by specifying this `dialectOptions: { ssl: true, }` in the `config.js` file under both test and development for it to work. You may want to check that.
> 
> I also get this when I ran `npm run seeders`
> 
> ```
> ERROR: Validation error
> npm ERR! code ELIFECYCLE
> npm ERR! errno 1
> npm ERR! express-authorshaven@1.0.0 seeders: `sequelize db:seed:all`
> npm ERR! Exit status 1
> ```
> 
> I'm presuming this might have to do with the seeder file

The error is with your local postgres configuration

**Comment by voke1** [comment]
> > Good work @debelistic However, I got this error when I ran `npm run test` `ERROR: no pg_hba.conf entry for host "129.56.74.94", user "uahtduhrsmfqxc", database "dfrvfn41s4u850", SSL off`.
> > I had to set ssl to true by specifying this `dialectOptions: { ssl: true, }` in the `config.js` file under both test and development for it to work. You may want to check that.
> > I also get this when I ran `npm run seeders`
> > ```
> > ERROR: Validation error
> > npm ERR! code ELIFECYCLE
> > npm ERR! errno 1
> > npm ERR! express-authorshaven@1.0.0 seeders: `sequelize db:seed:all`
> > npm ERR! Exit status 1
> > ```
> > 
> > 
> > I'm presuming this might have to do with the seeder file
> 
> The error is with your local postgres configuration

which local postgres configuration? I worked directly on your branch :)

**Comment by debelistic** [comment]
> > > Good work @debelistic However, I got this error when I ran `npm run test` `ERROR: no pg_hba.conf entry for host "129.56.74.94", user "uahtduhrsmfqxc", database "dfrvfn41s4u850", SSL off`.
> > > I had to set ssl to true by specifying this `dialectOptions: { ssl: true, }` in the `config.js` file under both test and development for it to work. You may want to check that.
> > > I also get this when I ran `npm run seeders`
> > > ```
> > > ERROR: Validation error
> > > npm ERR! code ELIFECYCLE
> > > npm ERR! errno 1
> > > npm ERR! express-authorshaven@1.0.0 seeders: `sequelize db:seed:all`
> > > npm ERR! Exit status 1
> > > ```
> > > 
> > > 
> > > I'm presuming this might have to do with the seeder file
> > 
> > 
> > The error is with your local postgres configuration
> 
> which local postgres configuration? I worked directly on your branch :)

`pg_hba.conf` should be a file or plugin for postgres configuration

File: src/bin/dev [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require('@babel/register');
+require('@babel/core');
+const http = require('http');
+const debug = require('debug');
+const app = require('../index').default;
```
Comment (misanj) [inline src/bin/dev:5]: Good job @debelistic  why do we have a dev file in the bin folder? It contains the exact same block of code in the www file except for line 5 that differs in path.

**Review by misanj — APPROVED** [review]
I now Understand what you where trying to do with the `dev` file and the `www` file after running the build script. for the `pg_hba.conf` error, the fix is to require postgres ssl in the `.env` file like this `PGSSLMODE=require`

I have tested this branch and it works fine.... well done @debelistic

**Comment by voke1** [comment]
**debelistic** 👍  works fine now

File: .envSample [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
-DATABASE_URL=
 DB_USER=
 DB_PASSWORD=
-DB_NAME=
 DB_HOST=
+DB_NAME=
+TEST_DB_NAME=
+TEST_DB_USER=
+TEST_DB_PASSWORD=
+TEST_DB_HOST=
```
Comment (victor-nach) [inline .envSample]: add a new line here.

File: package.json [hunk 3]
```diff
@@ -5,17 +5,17 @@
   "main": "index.js",
   "scripts": {
     "start": "node ./src/bin/www",
-    "devstart": "set DEBUG=dev && babel-watch ./src/bin/www",
+    "devstart": "set DEBUG=dev && babel-watch ./src/bin/dev",
     "build": "rm -rf dist && mkdir dist && babel src -s -d dist",
-    "test": "set NODE_ENV=test&& npm run reset:db && mocha -r @babel/register --recursive src/tests/ --exit --timeout 10000",
+    "test": " set NODE_ENV=test&& npm run reset-db&& mocha -r @babel/register --recursive src/tests/ --exit",
```
Comment (victor-nach) [inline package.json]: why did you remove the '--timeout 10000', due to the default timeout being too small, some of our tests would fail.

File: src/database/migrations/20190827125917-create-user.js [hunk 4]
```diff
@@ -13,7 +13,7 @@ module.exports = {
         notNull: true,
       },
     },
-    userName: {
+    username: {
```
Comment (victor-nach) [inline src/database/migrations/20190827125917-create-user.js]: I don't think you should remove the camel case here

File: src/models/user.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 export default (sequelize, DataTypes) => {
   const User = sequelize.define('User', {
-    userName: {
+    username: {
```
Comment (victor-nach) [inline src/models/user.js]: same here too, with the camel case

File: src/tests/routes/index.spec.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import chai from 'chai';
 import chaiHttp from 'chai-http';
-import app from '..';
+import app from '../..';
```
Comment (victor-nach) [inline src/tests/routes/index.spec.js]: why did you move the test file insde the tests / routes folder, do you intend for us to write tests separately for our controllers and models?

File: src/tests/routes/index.spec.js [hunk 7]
```diff
@@ -22,7 +23,8 @@ describe('GET /', () => {
   it('should show a welcome message for the home route', (done) => {
     chai.request(app)
       .get(`${endPoint}`)
-      .end((_err, res) => {
+      .end((err, res) => {
```
Comment (victor-nach) [inline src/tests/routes/index.spec.js]: if you're removing the _ from 'err', eslint would flag that, so you would also have to go and add a rule exception in the eslintrc file, or just stick with adding the underscore for unused variables.

File: src/tests/routes/index.spec.js (hunk 6 above)
Comment (debelistic) [reply src/tests/routes/index.spec.js]: That was how Anna specified it when set up the test, I was going to run it by you before I made the changes. I'm guess we'll be testing models and controllers, maybe

File: src/tests/routes/index.spec.js (hunk 7 above)
Comment (debelistic) [reply src/tests/routes/index.spec.js]: I added an if statement that makes use of the error, such that it would return err if there's an error with the request

File: src/tests/routes/index.spec.js (hunk 6 above)
Comment (debelistic) [reply src/tests/routes/index.spec.js]: That was the intruction from ttl, when I setup test

File: package.json (hunk 3 above)
Comment (debelistic) [reply package.json]: It will encourage longer response, if there's any response that's taking long, we should refactor. Besides 10s is long for response and I tested without the timeout and everything works fine

File: .envSample (hunk 2 above)
Comment (anneKay) [reply .envSample]: Please add a new line here

File: package.json (hunk 3 above)
Comment (anneKay) [reply package.json]: Any reason why you removed the timeout?

File: src/bin/dev (hunk 1 above)
Comment (anneKay) [reply src/bin/dev:5]: Please don't resolve reviews without attending to them or providing some answers to the questions

File: .envSample (hunk 2 above)
Comment (debelistic) [reply .envSample]: I just did

File: src/database/migrations/20190827125917-create-user.js (hunk 4 above)
Comment (anneKay) [reply src/database/migrations/20190827125917-create-user.js]: Any reason for this? If you would like to update columns names, that should be done with another migration file.

**Comment by anneKay** [comment]
Please don't resolve conversations without attending to the comments or providing answers/clarity to the questions

File: package.json (hunk 3 above)
Comment (anneKay) [reply package.json]: When we finally have several tests setup and running, it might timeout and cause tests to fail, adding this timeout limit doesn't necessarily mean that the tests must stay this duration before completing, please add this back.

File: src/validation/userValidation.js [hunk 8]
```diff
@@ -5,9 +5,9 @@ const userValidation = [
     .withMessage('First name field cannot be empty.'),
   check('lastName').trim().not().isEmpty()
     .withMessage('last name field cannot be empty.'),
-  check('userName').trim().not().isEmpty()
+  check('username').trim().not().isEmpty()
     .withMessage('username field cannot be empty.'),
-  check('userName').isAlphanumeric().withMessage('username should contain only letters and numbers'),
+  check('username').isAlphanumeric().withMessage('username should contain only letters and numbers'),
```
Comment (anneKay) [inline src/validation/userValidation.js]: Please return all these fields(userName) to what they were

**Review by victor-iheanacho — APPROVED** [review]

File: src/tests/routes/userSpec.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import chai from 'chai';
 import chaiHttp from 'chai-http';
-import app from '..';
+import app from '../..';
```
Comment (victor-iheanacho) [inline src/tests/routes/userSpec.js:3]: why did you move the tests into the routes folder?

**Review by victor-iheanacho — APPROVED** [review]

File: src/tests/routes/userSpec.js (hunk 9 above)
Comment (debelistic) [reply src/tests/routes/userSpec.js:3]: The tests for routes are meant to be in the folder, I think that's why ttl had us set the test folder up like that

File: src/validation/userValidation.js (hunk 8 above)
Comment (debelistic) [reply src/validation/userValidation.js]: We have dropped username field entirely from the user object

**Comment by debelistic** [comment]
> Please don't resolve conversations without attending to the comments or providing answers/clarity to the questions
I've provided answers to all questions on the PR
